design/custom-ca-support.md

```diff
@@ -0,0 +1,128 @@
+
+In order for Restic to be able to consume and use this cert, Velero will need
+the ability to write the CA bundle somewhere in memory for the Restic pod to
+consume it.
+
+To accomplish this, we can look at the code for managing restic repository
+credentials. The way this works today is that the key is stored in a secret in
+the Velero namespace, and each time Velero executes a restic command, the
+contents of the secret are read and written out to a temp file. The path to
+this file is then passed to restic and removed afterwards. pass the path of the
+temp file to restic, and then remove the temp file afterwards. See ref #1 and #2.
+
+This same approach can be taken for CA bundles. The bundle can be stored in a
+secret which is referenced on the BSL and written to a temp file prior to
+invoking Restic.
+
+[1](https://github.com/vmware-tanzu/velero/blob/master/pkg/restic/repository_manager.go#L238-L245)
+[2](https://github.com/vmware-tanzu/velero/blob/master/pkg/restic/common.go#L168-L203)
+
+## Detailed Design
+
+The `AWS_CA_BUNDLE` environment variable works for the Velero deployment
+because this environment variable is passed into the AWS SDK which is used in
+the [plugin][1] to build up the config object. This means that a user can
+simply define the CA bundle in the deployment as an env var. This can be
+utilized for the installation of Velero with a custom cert by simply setting
+this env var to the contents of the CA bundle, or the env var can be mapped to
+a secret which is controlled at installation time. I recommend using a secret
+as it makes the Restic integration easier as well.
+
+At installation time, if a user has specified a custom cert then the Restic
+daemonset should be updated to include the secret mounted at a predefined path.
+We could optionally use the system store for all custom certs added at
+installation time. Restic supports using the custom certs [in addition][3] to
+the root certs.
+
+In the case of the BSL being created with a secret reference, then at runtime
+the secret will need to be consumed. This secret will be read and applied to
+the AWS `session` object. The `getSession()` function will need to be updated
+to take in the custom CA bundle so it can be passed [here][4].
+
+The Restic controller will need to be updated to write the contents of the CA
+bundle secret out to a temporary file inside of the restic pod.The restic
+[command invocation][2] will need to be updated to include the path to the file
+as an argument to the restic server using `--cacert`. For the path when a user
+defines a custom cert on the BSL, Velero will be responsible for updating the
+daemonset to include the secret mounted as a volume at a predefined path.
+
+Where we mount the secret is a fine detail, but I recommend mounting the certs
+to `/certs` to keep it in line with the other volume mount paths being used.
+
+### Install command changes
+
+The installation flags should be updated to include the ability to pass in a
```

Comment:
@carlisia This is relevant to #2202. My guess would probably be part of the `plugin add` command there, with a new option for certs that would create the secret and do the updating on the deployment, and restic DaemonSet. I _think_ this is only valid for ObjectStore plugins, at least the restic DaemonSet part. I'm envisioning the command would take a path to the cert file, create the Secret in the cluster, then put a secretRef in the Velero Deployment/restic DaemonSet.

We'd also need to have the `backup-location (set|create)` commands have an argument to take a reference to a secret name (which is the cert) in the same namespace. Possibly the `snapshot-location (set|create)` commands, if it's relevant to them. The command would create or edit the BSL (or VSL) to have a reference to the Secret within the same namespace.

Reply:
Copy that! I'll add it to the doc, thank you.

Reply:
Seems it will work the same way as with the credentials for the provider (as far as adding it to the cluster).

Reply:
Yep, I think it'll work the same way, it's just a Secret with different contents.